pages/signup/register.tsx

```diff
@@ -1,60 +1,139 @@
 import React, { useState } from "react";
 import { Formik } from "formik";
-import { Button, TextInput, Label } from "@components/forms";
+import { Button, TextInput, Label, Alert, ErrorListItem } from "@components/forms";
 import { useAuth } from "@lib/hooks";
 import { useTranslation } from "next-i18next";
 import { GetServerSideProps } from "next";
 import { serverSideTranslations } from "next-i18next/serverSideTranslations";
 import { Confirmation } from "@components/auth/Confirmation/Confirmation";
+import * as Yup from "Yup";
+import { isValidGovEmail, isUpperCase, isLowerCase, isNumber, isSymbol } from "@lib/validation";
+import emailDomainList from "../../email.domains.json";
 
 export default function Register() {
   const { register } = useAuth();
   const [username, setUsername] = useState("");
   const { t } = useTranslation(["signup", "common"]);
+
+  const validationSchema = Yup.object().shape({
+    username: Yup.string()
+      .required(t("input-validation.required", { ns: "common" }))
+      .email(t("input-validation.email", { ns: "common" }))
+      .test(
+        "username-govEmail",
+        t("signUpRegistration.fields.username.error.validGovEmail"),
+        (value = "") => isValidGovEmail(value, emailDomainList.domains)
+      ),
+    firstName: Yup.string().required(t("input-validation.required", { ns: "common" })),
+    lastName: Yup.string().required(t("input-validation.required", { ns: "common" })),
+    password: Yup.string()
```

Comment:
I went with the sort of waterfall/filter approach with Yup of displaying one error at a time -vs- many at once with new lines or something like that

Reply:
I think thats okay for now. This will probably need changing as it might be super annoying for a user registering to keep failing on one condition at a time. However it fits the bill currently, we can probably work around this with a description for the password field containing the rules